Sources/Jetworking/Client.swift

```diff
@@ -0,0 +1,152 @@
+import Foundation
+
+enum APIError: Error {
+    case responseMissing
+    case decodingError
+}
+
+public final class Client {
+    // MARK: - Properties
+    private let clientConfiguration: ClientConfiguration
+
+    private lazy var session: URLSession = .init(
+        configuration: .default
+    )
+
+    // MARK: - Initialisation
+    init(clientConfiguration: ClientConfiguration) {
+        self.clientConfiguration = clientConfiguration
+    }
+
+    // MARK: - Methods
+    public func get<ResponseType>(endpoint: Endpoint<ResponseType>, _ completion: @escaping (Result<ResponseType, Error>) -> Void) {
+        let request: URLRequest = createRequest(forHttpMethod: .GET, andPathComponent: endpoint.pathComponent)
+        let dataTask = session.dataTask(with: request) { [weak self] data, urlResponse, error in
+            self?.handleResponse(data: data, urlResponse: urlResponse, error: error, endpoint: endpoint, completion: completion)
+        }
+
+        dataTask.resume()
+    }
+
+    public func post<BodyType: Encodable, ResponseType>(endpoint: Endpoint<ResponseType>, body: BodyType, _ completion: @escaping (Result<ResponseType, Error>) -> Void) {
+        let request: URLRequest = createRequest(forHttpMethod: .POST, andPathComponent: endpoint.pathComponent)
+        let bodyData: Data = try! clientConfiguration.encoder.encode(body) // TODO: remove force unwrap --> Error handling
+        let dataTask = session.uploadTask(with: request, from: bodyData) { [weak self] data, urlResponse, error in
+            self?.handleResponse(data: data, urlResponse: urlResponse, error: error, endpoint: endpoint, completion: completion)
+        }
+
+        dataTask.resume()
+    }
+
+    public func put<BodyType: Codable, ResponseType>(endpoint: Endpoint<ResponseType>, body: BodyType, _ completion: @escaping (Result<ResponseType, Error>) -> Void) {
```

Comment:
BodyType should only constraint to `Encodable`

Reply:
Done.